src/orion/service/client/experiment.py

```diff
@@ -0,0 +1,210 @@
+        space=None,
+        algorithms=None,
+        strategy=None,
+        max_trials=None,
+        max_broken=None,
+        storage=None,
+        branching=None,
+        max_idle_time=None,
+        heartbeat=None,
+        working_dir=None,
+        debug=False,
+        knowledge_base=None,
+        executor=None,
+    ):
+        """Instantiate an experiment using the REST API instead of relying on local storage"""
+
+        endpoint, token = storage["endpoint"], storage["token"]
+
+        workon = WorkonClientREST(endpoint, token)
+
+        experiment = workon.new_experiment(
+            name,
+            version=version,
+            space=space,
+            algorithms=algorithms,
+            strategy=strategy,
+            max_trials=max_trials,
+            max_broken=max_broken,
+            branching=branching,
+            max_idle_time=max_idle_time,
+            heartbeat=heartbeat,
+            working_dir=working_dir,
+            debug=debug,
+        )
+
+        client = ExperimentClientREST(
+            experiment,
+            executor=executor,
+            heartbeat=heartbeat,
+        )
+
+        storage = ClientActionREST(experiment, endpoint, token)
+
+        client.workon_client = workon
+        client.storage_client = storage
+        return client
+
+    # pylint: disable=super-init-not-called
+    def __init__(self, experiment, executor=None, heartbeat=None):
+        # Do not call super here; we do not want to instantiate the producer
+        self.workon_client = None
+
+        if heartbeat is None:
+            heartbeat = orion.core.config.worker.heartbeat
+
+        self._experiment = experiment
+        self.heartbeat = heartbeat
+        self._executor = executor
+        self._executor_owner = False
+        self.storage_client = None
+        self.plot = PlotAccessor(self)
+
+    def to_pandas(self, with_evc_tree=False):
+        raise RuntimeError()
+
+    @property
+    def _pacemakers(self):
+        # pylint: disable=protected-access
+        return self.workon_client._pacemakers
+
+    #
+    # Workon REST API overrides
+    #
+
+    @property
+    def is_broken(self):
+        """See `~ExperimentClient.is_broken`"""
+        try:
+            self.is_done
+        except BrokenExperiment:
+            return True
+        return False
+
+    @property
+    def is_done(self):
+        """See `~ExperimentClient.is_done`"""
+        return self.workon_client.is_done()
+
+    def suggest(self, pool_size=0) -> TrialCM:
+        """See `~ExperimentClient.suggest`"""
+        remote_trial = self.workon_client.suggest(pool_size=pool_size)
+        return remote_trial
+
+    def observe(self, trial, results, name: str = "objective"):
+        """See `~ExperimentClient.observe`"""
+        self.workon_client.observe(trial, results, name=name)
+
+    def release(self, trial, status="interrupted"):
+        """See `~ExperimentClient.release`"""
+        # we probably should not expose this
```

Comment:
Perhaps, but there should be a way to allow interupting a trial and so far `release` is the only way using the python API.